drivers/sensor/icm42688/icm42688.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ * Copyright (c) 2022 Esco Medical ApS
+ * Copyright (c) 2020 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/sys/byteorder.h>
+#include "icm42688.h"
+#include "icm42688_reg.h"
+#include "icm42688_spi.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ICM42688_LL, CONFIG_SENSOR_LOG_LEVEL);
+
+void icm42688_lock(const struct device *dev)
```

Comment:
Why do you have an empty lock function?

Reply:
It was used in Tom's additional changes. I'll remove for now until it's needed.